src/agl/WebAppWaylandAGL.cpp

```diff
@@ -16,8 +16,9 @@
 
 #include "WebAppWaylandAGL.h"
 #include "ApplicationDescription.h"
+#include "AglShell.h"
 
-WebAppWaylandAGL::WebAppWaylandAGL(const std::string& winType, std::shared_ptr<ApplicationDescription> desc)
-  : WebAppWayland(winType, desc->surfaceId(), desc->widthOverride(), desc->heightOverride(), kUndefinedDisplayId, "", desc->surfaceRole(), desc->panelType())
+WebAppWaylandAGL::WebAppWaylandAGL(const std::string& winType, std::shared_ptr<ApplicationDescription> desc, struct agl_shell_surface *surface)
```

Comment:
If the shell surfaces declaration is coming from the application config.xml... maybe we should just make the surfaces declaration part of the application description? (Even if we can still pass a different surfaces declaration from launching parameters).

Reply:
> maybe we should just make the surfaces declaration part of the application description?

Where exactly should this be? In the application itself? config.xml and everything around it are on their way out of so something similar or alternative to that when defining the surfaces roles would be needed. Choose config.xml for convenience.

Reply:
Ideally, to me it looks AGL code should populate ApplicationDescription using config.xml (adding new parameters as needed).